soda/scientific/soda/scientific/anomaly_detection/models/prophet_model.py

```diff
@@ -129,11 +129,14 @@ class PreprocessError(Exception):
 class ProphetDetector(BaseDetector):
     """ProphetDetector."""
 
+    integer_type_metrics = ["row_count", "duplicate_count", "missing_count", "invalid_count"]
```

Comment:
I'm a little curious about the reasoning behind this decision. Can you give a few pointers as to why we should only perform the rounding in the new way to integer metrics rather than to all?

Reply:
Rounding like this completely messes up the uncertainty interval, so I don't think it's something we should do unless necessary. This type of aggressive rounding could hurt us when we're predicting floats.

Thinking about I'll make a small adjustment for integers as well and only round if there's no integer between the upper and lower bound.

Reply:
I would add a TODO with the issue link to not forget to delete the hardcoded part.

Reply:
Just so you know, that didn't really answer my question. In other words why is it only necessary with ints? Why is it safer than with floats?

Reply:
It's necessary for row count because we need a way to interpret the prediction interval when it doesn't contain any integers. I.e. the confidence interval [10.5, 10.6] will classify any next observation as an anomaly if the next observation is an integer. So we need to round to [10, 11] (or do something else) to have a working implementation.

For floats that is not the case. Whatever your prediction interval looks like there's always a next observation that can be contained by it.

Also imagine that you have the values 14.2, 10.55, 10.65, 10.2, 10.35, 15.2, 10.2, 10.9, 11.1 leading to a prediction interval of [10.5, 10.6] (didn't do the math just an example). If you round downwards and upwards you get the confidence interval [10, 11] which is not sensitive to the small difference in magnitude of the previous values.

Reply:
Ok that all makes sense now. Thanks for the explanation. So for integers I think this is totally legit. 

That being said, I'm not sure I love the idea of listing the metrics here. We'll always fall behind + it mixes knowledge from another domain (core in that case). Do we have a way to simply check whether the actual measurements time series is made of integers. I forgot if that would be the case but if so, I'd prefer that we go with that. If we must rely on the metric, I think that actually we will want to work with core so that they can communicate about the metric's type. I think it should be ok to get this from them and have cloud be ok to store it. In fact, it could help for a lot of other things. More than happy to look into putting up this series of feature requests with the teams if we deem it necessary.

Right now, though, if listing the metrics is our ownly option. Please add a `# TODO` comment that refers to a JIRA issue that needs to be created to refactor this when/if cloud and core agree to support datatype info storage.

Reply:
@bastienboutonnet No I don’t think so I’m afraid, also see my [comment to Baturay](https://github.com/sodadata/soda-core/pull/1780#issuecomment-1415481250). The reason is that we use interpolation between values so row counts can be floats. Looking at both the historical check results and historic measurements there seems to be no field that indicates type.